pkg/controller/registry/reconciler/grpc.go

```diff
@@ -168,6 +168,62 @@ func (c *GrpcRegistryReconciler) ensureService(source grpcCatalogSourceDecorator
 	return err
 }
 
+// updateRegistryPodByDigest is an internal method that verifies the latest catalog source by comparing image digests.
+// If the digests do not match (i.e. there is a new version of the catalog source) then overwrite the catalog source pod.
+// This is done to only have the container runtime (CRI-O) talk to the container registry.
+func (c *GrpcRegistryReconciler) updateRegistryPodByDigest(source grpcCatalogSourceDecorator) bool {
+	podDigestChan := make(chan string, 1)
+
+	go c.getPodDigest(source, podDigestChan)
+
+	select {
+	case newCatalogSourceImageDigest := <-podDigestChan:
+		if newCatalogSourceImageDigest == source.Pod().Status.ContainerStatuses[0].ImageID {
+			return false
+		}
+		// we have a new imageID in our catalog source container: a new version of the catalog source image exists
+		// update catalog source pod
+		return true
+	default:
+		return false
+	}
+}
+
+// getPodDigest is an internal method that creates a pod using the latest catalog source.
+// Once the pod comes up, it puts the container image digest onto a channel.
+func (c *GrpcRegistryReconciler) getPodDigest(source grpcCatalogSourceDecorator, podChan chan string) {
+	// remove label from pod to ensure service does accidentally route traffic to the pod
+	p := source.Pod()
+	p.Labels[CatalogSourceLabelKey] = ""
```

Comment:
When we remove the labels here, to prevent the service routing to this pod, we also make the pod deletion in `ensurePod` not pick up this new pod - it looks at the labels to figure out which pod to delete. So we would need an additional label or mechanism to ensure these pods get deleted properly